clusterversion: remove redundant version gates introduced in 23.2

yuzefovich opened the pull request:
This commit removes multiple versions introduced in 23.2 cycle that
don't require an upgrade by replacing them with `V23_2`. The
following gates are removed:
- `V23_2TTLAllowDescPK`
- `V23_2_PartiallyVisibleIndexes`
- `V23_2_Procedures`
- `V23_2_PLpgSQL`
- `V23_2_UDFMutations`.

Epic: None

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/112873)
<!-- Reviewable:end -->

yuzefovich reviewed (commented):
Note that there appear to be some other version gates below SQL that don't have corresponding upgrades, but I didn't touch those.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/112873#-:-NhSdCvW2Cj6cTsoLx1p:bm1k09v)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt and @mgartner)



<!-- Sent from Reviewable.io -->

DrewKimball reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Thanks for doing this!

> Reviewed 14 of 14 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/112873#-:-NhSfiI_Em3LoU0PbnVO:b-pt8m11)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dt and @mgartner)



<!-- Sent from Reviewable.io -->

dt commented:
the renumbering of the later versions to eliminate the gap isn't necessary and would be forbidden if we'd changed 23.2 branch to use non-dev versions, but we haven't done so yet so I guess it is okay. That said, gaps are fine and have no effect: we step through each version returned by `listBetween` during an upgrade, but that just means we step through each _defined_ version it is is start <= version < end, so simply removing from the list is enough to mean we won't do a fence/wait/advance/wait dance for it.

dt reviewed (approved):
Personally I wouldn't re-number the unrelated versions, but I don't have a strong reason not to do so either.

yuzefovich committed:
clusterversion: remove redundant version gates introduced in 23.2

This commit removes multiple versions introduced in 23.2 cycle that
don't require an upgrade by replacing them with `V23_2`. The
following gates are removed:
- `V23_2TTLAllowDescPK`
- `V23_2_PartiallyVisibleIndexes`
- `V23_2_Procedures`
- `V23_2_PLpgSQL`
- `V23_2_UDFMutations`.

Epic: None

Release note: None

yuzefovich commented:
TFTRs!

bors r+
